posts/2021-03-26-MP-JWT-1.2.adoc

```diff
@@ -0,0 +1,290 @@
+- <<examples, Examples>>
+    * <<accepting-json-web-encryption-jwe-tokens, Accepting JSON Web Encryption (JWE) tokens>>
+    * <<accepting-jwts-in-cookies, Accepting JWTs in cookies>>
+
+[#new-mp-config-properties]
+== New MP Config properties
+
+Version 1.2 of the MicroProfile JWT specification adds the following MicroProfile Config properties to control different aspects of JWT validation.
+
+- <<mp-jwt-token-header, mp.jwt.token.header>>
+- <<mp-jwt-token-cookie, mp.jwt.token.cookie>>
+- <<mp-jwt-verify-audiences, mp.jwt.verify.audiences>>
+- <<mp-jwt-decrypt-key-location, mp.jwt.decrypt.key.location>>
+- <<mp-jwt-verify-publickey-algorithm, mp.jwt.verify.publickey.algorithm>>
+
+*Note:* If you do not wish to use MicroProfile Config properties, the `<mpJwt>` element in Open Liberty's server.xml provides equivalent functionality for each of these properties. Whenever there is an equivalent server.xml attribute for an MP Config property, the server.xml attribute takes precedence if configured.
+
+[#mp-jwt-token-header]
+=== mp.jwt.token.header
+
+The `mp.jwt.token.header` property allows you to control the HTTP request header that is expected to contain a JWT. You can specify either `"Authorization"` (default) or `"Cookie"`.
+
+The token header can also be configured in the `tokenHeader` attribute of the `<mpJwt>` server.xml element:
+[source,xml]
+----
+<mpJwt id="myMpJwt" tokenHeader="Authorization" ... />
+----
+
+[#mp-jwt-token-cookie]
+=== mp.jwt.token.cookie
+
+The `mp.jwt.token.cookie` property allows you to specify the name of the cookie that is expected to contain a JWT. The default value is `"Bearer"` if not specified.
+
+The token cookie name can also be configured in the `cookieName` attribute of the `<mpJwt>` server.xml element as shown here:
+[source,xml]
+----
+<mpJwt id="myMpJwt" tokenHeader="Cookie" cookieName="Bearer" ... />
+----
+
+*Note:* The `mp.jwt.token.header` MP Config property or `tokenHeader` server.xml attribute must be set to `"Cookie"` for the `mp.jwt.token.cookie` property or `cookieName` attribute to be recognized.
+
+[#mp-jwt-verify-audiences]
+=== mp.jwt.verify.audiences
+
+The `mp.jwt.verify.audiences` property allows you to create list of allowable audience values. At least one of these must be found in the `"aud"` claim of the JWT. Previously, this had to be configured in the server.xml file. Now you can configure the audiences in the MicroProfile Config property as follows:
+[source]
+----
+mp.jwt.verify.audiences=conferenceService, adminService
+----
+
+[#mp-jwt-decrypt-key-location]
+=== mp.jwt.decrypt.key.location
+
+To support JSON Web Encryption tokens, the `mp.jwt.decrypt.key.location` property allows you to specify the location of the Key Management Key. This is the private key that is used to decrypt the Content Encryption Key, which is then used to decrypt the JWE ciphertext. The private key must correspond to the public key that is used to encrypt the Content Encryption Key.
```

Comment:
shouldn't "Key Management Key" be "Private Management Key"

Reply:
"Key Management Key" is the technical term from the JWE specification and we want to be specific as to what Key, from the specification, this is referring to.

Reply:
"Key management key" is the noun phrase we've used throughout our docs. The [JWE spec](https://tools.ietf.org/html/rfc7516#section-2) defines the phrase "key management mode" as the "method of determining the Content Encryption Key value to use". Building off of that, we decided the key used to wrap the Content Encryption Key would then be the key management key. Our `<jwtBuilder>` and `<mpJwt>` elements have corresponding `keyManagementKeyAlias` attributes.

Reply:
ah, thanks for the explanation.  Key Management Key sounded odd to me, like a mistake, and when I searched for jwt.decrypt.key.location, I saw that smallrye seemed to refer to it as the "Private Decryption Key" and in an [MP issue](https://github.com/eclipse/microprofile-jwt-auth/issues/193) it was also referenced to as Private Decryption Key.